src/host/_output.cpp

```diff
@@ -254,76 +394,40 @@ void WriteToScreen(SCREEN_INFORMATION& screenInfo, const Viewport& region)
-        ImageSlice::EraseCells(screenInfo.GetTextBuffer(), startingCoordinate, cellsModified);
-
-        // Notify accessibility
-        if (screenInfo.HasAccessibilityEventing())
-        {
-            auto endingCoordinate = startingCoordinate;
-            bufferSize.WalkInBounds(endingCoordinate, cellsModifiedCoord);
-            screenInfo.NotifyAccessibilityEventing(startingCoordinate.x, startingCoordinate.y, endingCoordinate.x, endingCoordinate.y);
-        }
-
-        // GH#3126 - This is a shim for powershell's `Clear-Host` function. In
-        // the vintage console, `Clear-Host` is supposed to clear the entire
-        // buffer. In conpty however, there's no difference between the viewport
-        // and the entirety of the buffer. We're going to see if this API call
-        // exactly matched the way we expect powershell to call it. If it does,
-        // then let's manually emit a ^[[3J to the connected terminal, so that
-        // their entire buffer will be cleared as well.
         auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
-        if (enablePowershellShim && gci.IsInVtIoMode())
+        if (const auto io = gci.GetVtIo(&OutContext))
         {
-            const auto currentBufferDimensions{ screenInfo.GetBufferSize().Dimensions() };
-
-            const auto wroteWholeBuffer = lengthToWrite == (currentBufferDimensions.area<size_t>());
-            const auto startedAtOrigin = startingCoordinate == til::point{ 0, 0 };
-            const auto wroteSpaces = character == UNICODE_SPACE;
-
-            if (wroteWholeBuffer && startedAtOrigin && wroteSpaces)
+            // GH#3126 - This is a shim for powershell's `Clear-Host` function. In
+            // the vintage console, `Clear-Host` is supposed to clear the entire
+            // buffer. In conpty however, there's no difference between the viewport
+            // and the entirety of the buffer. We're going to see if this API call
+            // exactly matched the way we expect powershell to call it. If it does,
+            // then let's manually emit a Full Reset (RIS).
+            if (enablePowershellShim)
             {
-                // It's important that we flush the renderer at this point so we don't
-                // have any pending output rendered after the scrollback is cleared.
-                ServiceLocator::LocateGlobals().pRender->TriggerFlush(false);
-                hr = gci.GetVtIo()->ManuallyClearScrollback();
+                const auto currentBufferDimensions{ OutContext.GetBufferSize().Dimensions() };
+                const auto wroteWholeBuffer = lengthToWrite == (currentBufferDimensions.area<size_t>());
+                const auto startedAtOrigin = startingCoordinate == til::point{ 0, 0 };
+                const auto wroteSpaces = character == UNICODE_SPACE;
+
+                if (wroteWholeBuffer && startedAtOrigin && wroteSpaces)
+                {
+                    WriteCharsVT(OutContext, L"\033c");

```

Comment:
I'm not sure RIS is correct here - Linux for example separates `clear` and `reset`...

Reply:
I felt like RIS is a better choice exactly because we don't have separate commands for the two. If we don't use RIS for `cls`, how else would you get a "broken" terminal back into a working state as someone who doesn't know VT too well?

Reply:
I'm with @DHowett on this. Using `RIS` for `cls` is a breaking change, because it'll nuke your active attributes (amongst other things). If we want `cls` to fix broken terminal state, it might make sense to reset certain VT-specific settings (things like margins, some modes, the alt buffer, etc.), but a hard reset seems like a step too far.

Reply:
I missed your comment. That's already 2 people against it, so I suppose I should change it.